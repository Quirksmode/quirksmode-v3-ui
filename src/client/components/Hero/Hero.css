.Hero {
  height: 300px;
  position: relative;

  @media (--mobile-landscape) {
    height: 400px;
  }

  @media (--tablet-portrait) {
    height: 500px;
  }

  &::before {
    background: var(--color-black);
    bottom: 0;
    content: '';
    height: 100%;
    left: 0;
    opacity: .5;
    position: absolute;
    right: 0;
    top: 0;
    width: 100%;
  }

  &__img {
    height: 100%;
    max-width: none;
    object-fit: cover;
    object-position: 50% 0;
    text-align: center;
    width: 100%;
  }

  &__inner {
    align-items: flex-end;
    display: flex;
    height: 100%;
    left: 0;
    position: absolute;
    right: 0;
    top: 0;
  }

  &__textWrap {
    margin-bottom: 20px;
    max-width: 650px;
    position: relative;
    width: 100%;

    @media (--mobile-landscape) {
      margin-bottom: 40px;
    }

    @media (--tablet-portrait) {
      margin-bottom: 60px;
    }
  }

  &__textWrapTitle {
    color: var(--color-white);
    font-size: 4.5rem;
    line-height: 4rem;

    @media (--mobile-landscape) {
      font-size: 5rem;
      line-height: 5rem;
    }

    @media (--tablet-portrait) {
      font-size: 6rem;
      line-height: 5.2rem;
    }
  }

  h3 {
    color: var(--color-white);
    font-weight: normal;
  }

  .Breadcrumbs,
  &__share {
    top: 20px;

    @media (--mobile-landscape) {
      top: 30px;
    }

    @media (--tablet-portrait) {
      top: 40px;
    }
  }

  .Breadcrumbs {
    background: rgba(0, 0, 0, .4);
    border-radius: 3px;
    left: 0;
    margin: 0;
    padding: 5px 10px;
    position: absolute;

    svg {
      fill: var(--color-white);
      height: 20px;
      transition: fill .2s ease-out;
      width: 20px;

      &:hover {
        .no-touch & {
          fill: var(--color-qm-blue);
        }
      }
    }

    a,
    span {
      color: var(--color-white);
      font-size: 1.4rem;
      line-height: 1.4rem;
      transition: color .2s ease-out;

      &:hover {
        .no-touch & {
          color: var(--color-qm-blue);
        }
      }
    }
  }

  &__share {
    position: absolute;
    right: 0;
  }
}
