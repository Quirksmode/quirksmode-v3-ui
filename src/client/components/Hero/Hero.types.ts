import { ReactNode } from 'react';
import { Hero } from 'client/pages/PagePortfolioSingle/PagePortfolioSingle.types';

export interface HeroProps {
  hero: Hero;
  title: string;
  subtitle?: string;
  url: string;
  type: 'portfolio' | 'blog';
}
