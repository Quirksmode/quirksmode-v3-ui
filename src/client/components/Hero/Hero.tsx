import React from 'react';
import { NavLink } from 'react-router-dom';
import Breadcrumbs from 'components/Breadcrumbs/Breadcrumbs';
import Share from 'components/Share/Share';
import { HeroProps } from './Hero.types';

/**
 * Hero Component
 */
const Hero: React.FC<HeroProps> = ({
  hero,
  title,
  subtitle = '',
  url,
  type,
}) => (
  <section className="Hero">
    <picture>
      <source
        type="image/webp"
        srcSet={`${hero.image.sizes.fullWidth1800}.webp 1x,
                ${hero.image.sizes.fullWidth1800_2x}.webp 1.5x,
                ${hero.image.sizes.fullWidth1800_2x}.webp 2x`}
        media="(min-width: 1024px)"
      />
      <source
        type="image/jpeg"
        srcSet={`${hero.image.sizes.fullWidth1800} 1x,
                ${hero.image.sizes.fullWidth1800_2x} 1.5x,
                ${hero.image.sizes.fullWidth1800_2x} 2x`}
        media="(min-width: 1024px)"
      />
      <source
        type="image/webp"
        srcSet={`${hero.image.sizes.fullWidth1024}.webp 1x,
                ${hero.image.sizes.fullWidth1024_2x}.webp 1.5x,
                ${hero.image.sizes.fullWidth1024_2x}.webp 2x`}
        media="(min-width: 768px)"
      />
      <source
        type="image/jpeg"
        srcSet={`${hero.image.sizes.fullWidth1024} 1x,
                ${hero.image.sizes.fullWidth1024_2x} 1.5x,
                ${hero.image.sizes.fullWidth1024_2x} 2x`}
        media="(min-width: 768px)"
      />
      <source
        type="image/webp"
        srcSet={`${hero.image.sizes.fullWidth768}.webp 1x,
                ${hero.image.sizes.fullWidth768_2x}.webp 1.5x,
                ${hero.image.sizes.fullWidth768_2x}.webp 2x`}
        media="(min-width: 481px)"
      />
      <source
        type="image/jpeg"
        srcSet={`${hero.image.sizes.fullWidth768} 1x,
                ${hero.image.sizes.fullWidth768_2x} 1.5x,
                ${hero.image.sizes.fullWidth768_2x} 2x`}
        media="(min-width: 481px)"
      />
      <source
        type="image/webp"
        srcSet={`${hero.image.sizes.fullWidth481}.webp 1x,
                ${hero.image.sizes.fullWidth481_2x}.webp 1.5x,
                ${hero.image.sizes.fullWidth481_2x}.webp 2x`}
      />
      <img
        className="Hero__img"
        srcSet={`${hero.image.sizes.fullWidth481} 1x,
                ${hero.image.sizes.fullWidth481_2x} 1.5x,
                ${hero.image.sizes.fullWidth481_2x} 2x`}
        src={hero.image.sizes.fullWidth481}
        alt={hero.image.alt}
        data-test="Hero__img"
      />
    </picture>
    <div className="Hero__inner grid">
      <div
        className={`Hero__textWrap grid__content${
          hero.alignCenter ? ' grid__content--center' : ''
        }`}
        data-test="Hero__textWrap"
      >
        <h1 data-test="Hero__textWrapTitle" className="Hero__textWrapTitle">
          {title}
        </h1>
        <h3 data-test="Hero__subtitle">{subtitle}</h3>
      </div>
      <Breadcrumbs>
        <span className="Breadcrumbs__divider">&gt;</span>
        <NavLink to={`/${type}`} className="Breadcrumbs__link">{` ${
          type === 'blog' ? 'Blog' : 'Portfolio'
        } `}</NavLink>
      </Breadcrumbs>
      <div className="Hero__share">
        <Share title={title} url={url} />
      </div>
    </div>
  </section>
);

export default Hero;
