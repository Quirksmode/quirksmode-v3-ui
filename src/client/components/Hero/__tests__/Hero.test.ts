import { findByTestAttr, mountRender, reduxWrap } from 'client/tests/testUtils';
import Hero from '../Hero';
import 'tests/mocks/intersectionObserver';

describe('Hero', () => {
  let wrapper: any;
  const props = {
    hero: {
      image: {
        alt: 'Hero Image Alt Test',
        sizes: {
          fullWidth481:
            'https://cms.quirksmode.co.uk/wp-content/uploads/2020/01/smashingconf-barcelona-neckinfo-1-481x200.jpg',
          fullWidth481_2x:
            'https://cms.quirksmode.co.uk/wp-content/uploads/2020/01/smashingconf-barcelona-neckinfo-1-962x401.jpg',
          fullWidth768:
            'https://cms.quirksmode.co.uk/wp-content/uploads/2020/01/smashingconf-barcelona-neckinfo-1-768x320.jpg',
          fullWidth768_2x:
            'https://cms.quirksmode.co.uk/wp-content/uploads/2020/01/smashingconf-barcelona-neckinfo-1-1536x640.jpg',
          fullWidth1024:
            'https://cms.quirksmode.co.uk/wp-content/uploads/2020/01/smashingconf-barcelona-neckinfo-1-1024x427.jpg',
          fullWidth1024_2x:
            'https://cms.quirksmode.co.uk/wp-content/uploads/2020/01/smashingconf-barcelona-neckinfo-1.jpg',
          fullWidth1800:
            'https://cms.quirksmode.co.uk/wp-content/uploads/2020/01/smashingconf-barcelona-neckinfo-1-1800x750.jpg',
          fullWidth1800_2x:
            'https://cms.quirksmode.co.uk/wp-content/uploads/2020/01/smashingconf-barcelona-neckinfo-1.jpg',
        },
      },
      alignCenter: true,
    },
    title: 'Smashing Conference 2015 - Barcelona',
    url:
      'https://cms.quirksmode.co.uk/blog/smashing-conference-2015-barcelona/',
    subtitle: '15th November 2015',
    type: 'project',
  };

  beforeEach(() => {
    wrapper = mountRender(reduxWrap(Hero, props));
  });

  it('matches snapshot', () => {
    expect(wrapper.html).toMatchSnapshot();
  });

  it('should display title', () => {
    const title = findByTestAttr(wrapper, 'Hero__textWrapTitle');
    expect(title.text()).toBe(props.title);
  });

  it('should display subtitle', () => {
    const subtitle = findByTestAttr(wrapper, 'Hero__subtitle');
    expect(subtitle.text()).toBe(props.subtitle);
  });

  it('should display image', () => {});

  it('should apply grid__content--center class when alignCenter is true', () => {});

  it('should show breadcrumbs', () => {});

  it('should display the correct Type', () => {});

  it('should show share links', () => {});
});
