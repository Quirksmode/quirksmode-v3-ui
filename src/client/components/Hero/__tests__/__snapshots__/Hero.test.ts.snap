// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Hero matches snapshot 1`] = `[Function]`;
